bin/ingester.ts

```diff
@@ -16,21 +16,31 @@ const stream = pretty({
 
 dotenv?.config();
 
-
 const directoryUrl = process.env.DIRECTORY_URL;
 if (!directoryUrl) {
     throw new Error("DIRECTORY_URL environment variable is not set");
 }
 
 export const logger = pino({
-    name: 'InfluxDB Sparkplug Ingester'
+    name: 'InfluxDB Sparkplug Ingester',
+    level: process.env.LOG_LEVEL || 'info',
 }, stream);
 
 
 const client = await new ServiceClient({
     directory_url: directoryUrl,
 }).init();
 
+// Overwrite MQTT server if specified in environment
+if (process.env.MQTT_URL) {
+    client.Discovery.set_service_url(UUIDs.Service.MQTT, process.env.MQTT_URL);
+}
```

Comment:
If you pass `env: process.env` to the ServiceClient constructor it will already do this. (You don't need any other arguments, it pulls `DIRECTORY_URL` from the env too.) `CMD_ESC_URL` is not handled yet but that would be an easy patch to `utilities`.

Maybe Discovery needs a bit of refactoring to support pulling URLs from a supplied environment as well as from the Directory.